umbraco/Umbraco-CMS
# V15: Document Type Create Options

### Description

Creates an `entityCreateOptionAction` extension for each of the Document Type create actions. This builds on the work started in https://github.com/umbraco/Umbraco.CMS.Backoffice/pull/2509, and replaces the existing "Document Type Create Options modal" (which has hardcoded options).

To align the UI/UX consistency for the existing Create Options, amends were required to the `<umb-ref-item>` component, (refactored to extend `uui-ref-node` rather than duplicate code/functionality). This mostly meant updating existing usage to add the `select-only` attribute.

// @madsrasmussen Tagging you in, as you may wish to cast an eye on these amends.

### How to test?

Go to the Settings section, try creating a new document type:

- Does the create modal appear?
- Does it have any options?
- Do they work?
